site/app/controllers/NavigationController.php

```diff
@@ -19,7 +19,7 @@ public function navigationPage() {
             $gradeables_list = new GradeableList($this->core);
         }
         catch (DatabaseException $e) {
-            ExceptionHandler::handleException($e);
+            ExceptionHandler::handleException($e, $this->core->getUser()->getId());
```

Comment:
What happens if `getUser()` returns `null`?

Reply:
realistically the only time this would happen is if authentication has an error, but I added a null check just in case so it returns null. In ExceptionHandler there is a null check.

Reply:
Can you add the null-safe operator here too just to be safe. Referring to the line I commented on.

Reply:
sorry added it to another place that also made sense